client/src/Components/Layouts/ShoppingCart/ShoppingCart.js

```diff
@@ -0,0 +1,141 @@
+import React ,{useEffect,useState} from 'react'
+import Typography from '@material-ui/core/Typography'
+import Button from '@material-ui/core/Button'
+import BottomNavigation from '@material-ui/core/BottomNavigation'
+import SecondHeader from '../../Common/SecondHeder/SecondHeader'
+import useStyles from './style'
+import axios from 'axios'
+import Card from '@material-ui/core/Card'
+import CardContent from '@material-ui/core/CardContent'
+import CardMedia from '@material-ui/core/CardMedia'
+import IconButton from '@material-ui/core/IconButton'
+import DeleteIcon from '@material-ui/icons/Delete'
+import { useHistory } from 'react-router-dom'
+
+const ShoppingCart = () =>{
+    const classes = useStyles()
+    const [cartList,setCartList]=useState(null)
+    const [clientId,setId]=useState(null)
+    const history = useHistory()
+
+    useEffect(
+      ()=>{      
+        axios.get('/api/user-id')
+          .then(result=> {
+            if(!result.data.success){
+             history.push('/login')
+            }          
+            else{ 
+              if(!cartList){
+              setId(result.data.id)
+             return axios.get(`/api/cart/${clientId}`)
+              .then(result => {
+                  if(result.data.length===0){
+                        return 
+                  }
+                  setCartList(result.data)
+              })
+              .catch(err => err)}
+            }
+          })
+          .catch(err=>err)
+        
+        
+        
+        },[cartList,clientId]
+    )
+    const priceCounter=()=>{
+        let counter =0
+        for(let i=0;i< cartList.length;i++){
+             counter= counter+cartList[i].price
+        } 
+        return counter
+    }
+    const deleteItem = (itemId)=>{
+
+        axios.delete('/api/cart/delete',
+       { data: {
+            artwork: itemId,
+            user : clientId
+        }})
+        window.location.reload()
```

Comment:
use `history.go()` to rerender the component rather than a full page reload.

Reply:
isn't the same thing `reload` or `.go('/cart')`?

Reply:
`history.go()` rerenders the same component. `.go('/cart')` might be the same thing but I've never used it. However, `location.reload` unmounts the entire app and remounts it. So the first two are within the same update phase of the lifecycle, while the third one will recreate the entire lifecycle from phase 1 (mounting) which is a little heavier on the computer.